pkg/exporter/process.go

```diff
@@ -101,16 +101,22 @@ func InitExportingProcess(collectorAddr net.Addr, obsID uint32, tempRefTimeout u
 	return expProc, nil
 }
 
-func (ep *ExportingProcess) AddRecordAndSendMsg(recType entities.ContentType, rec entities.Record) (int, error) {
-	if recType == entities.Template {
-		ep.updateTemplate(rec.GetTemplateID(), rec.GetTemplateElements(), rec.GetMinDataRecordLen())
-	} else if recType == entities.Data {
-		err := ep.dataRecSanityCheck(rec)
-		if err != nil {
-			return 0, fmt.Errorf("AddRecordAndSendMsg: error when doing sanity check:%v", err)
+func (ep *ExportingProcess) AddRecordAndSendMsg(recType entities.ContentType, records ...entities.Record) (int, error) {
```

Comment:
AddRecordsAndSendMsg. Or this API can be modified a bit to make it AddSetsAndSendMsg as there is object for Set now. I prefer the latter option. It is better to follow the abstraction: msg->sets->records. What do you think?

Reply:
Agreed. Will do the corresponding changes. Thanks!